ncm-ssh/src/main/pan/components/ssh/schema.pan

```diff
@@ -10,26 +10,72 @@ include 'pan/types';
 
 type ssh_preferred_authentication = string with match(SELF, '^(gssapi-with-mic|hostbased|publickey|keyboard-interactive|password)$');
 
+type ssh_ciphers = string with is_valid_ssh_cipher(SELF);
+type ssh_hostkeyalgorithms = string with match(SELF, "^(ssh-(rsa|dss|ed25519)|ecdsa-sha2-nistp(256|384|521)|(ssh-rsa-cert-v01|ssh-dss-cert-v01|ecdsa-sha2-nistp256-cert-v01|ecdsa-sha2-nistp384-cert-v01|ecdsa-sha2-nistp521-cert-v01|ssh-rsa-cert-v00|ssh-dss-cert-v00|ssh-ed25519-cert-v01)@openssh.com)$");
+type ssh_kbdinteractivedevices = string with match (SELF, "^(bsdauth|pam|skey)$");
+type ssh_kexalgorithms = string with match (SELF, "^(diffie-hellman-group-exchange-sha256|ecdh-sha2-nistp(256|384|521)|curve25519-sha256@libssh.org)$");
+type ssh_MACs = string with is_valid_ssh_MAC(SELF);
+
+function is_valid_ssh_MAC = {
+    match(ARGV[0], "^(hmac-(sha1|sha2-256|sha2-512|ripemd160)|(hmac-ripemd160|umac-64|umac-128|hmac-sha1-etm|hmac-sha2-256-etm|hmac-sha2-512-etm|hmac-ripemd160-etm|umac-64-etm|umac-128-etm)@openssh.com)$");
```

Comment:
no need to specify hmac-ripemd160, already matched by hmac-(..|ripemd160)
also add ``(-etm)?`` to the first part of the regex to get rid of all the seperate -etm entries.
Same can be done for the umac entries